### PR TITLE
Select 508 enchancement

### PR DESCRIPTION
**Pull request by zacharymorel**
## Description
Closes [19157](https://github.com/department-of-veterans-affairs/va.gov-team/issues/19157);
OG ticket: [19117](https://github.com/department-of-veterans-affairs/va.gov-team/issues/19117#issuecomment-768406699)

## Testing done
Added unit test, ran unit tests, ran axe chrome extensions locally, manual testing with voice over screen reader active.

## Screenshots
Screen Shot 2021-01-29 at 11 06 09 AM

## Acceptance criteria
- [x] Announce change in select state to screen reader users.

## Definition of done
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: src/components/Select/Select.jsx [hunk 1]
```diff
@@ -31,9 +31,9 @@ class Select extends React.Component {
       errorSpanId = `${this.selectId}-error-message`;
       errorSpan = (
         <span
-          className="usa-input-error-message"
+          className='usa-input-error-message'
```
Comment (zacharymorel) [inline src/components/Select/Select.jsx:34]: I had prettier on by default, I _undid_ a lot of the changes made on save (you can see in commmits). There's a few left over like this.

File: src/components/Select/Select.stories.jsx [hunk 2]
```diff
@@ -50,3 +50,9 @@ NoBlankOption.args = {
   ...defaultArgs,
   includeBlankOption: false,
 };
+
+export const AriaLiveRegion = Template.bind({});
+AriaLiveRegion.args = {
+  ...defaultArgs,
+  ariaLiveRegionText: 'The following branch was selected: ',
+};
```
Comment (zacharymorel) [inline src/components/Select/Select.stories.jsx:58]: I am not opinionated about Prop Names, if y'all have something different, let me know!

**Review by 1Copenut — CHANGES_REQUESTED** [review]

File: src/components/Select/Select.jsx [hunk 3]
```diff
@@ -83,6 +83,9 @@ class Select extends React.Component {
         {errorSpan}
         <select
           className={this.props.additionalClass}
+          aria-controls={
```
[the remark this answers is not in the document]
Comment (zacharymorel) [reply src/components/Select/Select.jsx:86]: Thanks for the feedback! I'll give this a go in a bit here and test locally. Should see a change soon.

[the remark this answers is not in the document]
Comment (zacharymorel) [reply src/components/Select/Select.jsx]: Updated!

**Review by 1Copenut — APPROVED** [review]
👍 I ran through the Storybook output with the following pairings:

* MacOS + Safari + VO
* Win10 + Edge + JAWS 21
* Win10 + Chrome + JAWS 21
* Win10 + IE11 + JAWS 21
* Win10 + Edge + NVDA
* Win10 + Chrome + NVDA
* Win10 + Firefox + NVDA

All worked as expected except IE11 + JAWS. I'm not certain what we can do to make IE work as expected. That browser has some challenges with live regions.

/cc @joshkimux

File: src/components/Select/Select.jsx (hunk 1 above)
Comment (cvalarida) [reply src/components/Select/Select.jsx:34]: Yeah, we still need to add an `.eslintrc` to this repo. :sweat_smile:

File: src/components/Select/Select.stories.jsx (hunk 2 above)
Comment (cvalarida) [reply src/components/Select/Select.stories.jsx:58]: This looks great! It's very clear what this is doing. :+1:

**Review by cvalarida — APPROVED** [review]
Thanks for the improvement! :raised_hands:

**Pull request merged**